StardewArchipelago/GameModifications/Modded/JunimoShopGenerator.cs

```diff
@@ -0,0 +1,504 @@
+        private PersistentStock BluePersistentStock { get; }
+        private PersistentStock GreyPersistentStock { get; }
+        private PersistentStock YellowPersistentStock { get; }
+        private PersistentStock RedPersistentStock { get; }
+        private PersistentStock OrangePersistentStock {get; }
+        private static Dictionary<string, JunimoVendor> JunimoVendors { get; set; }
+        private Dictionary<int, int> BlueItems { get; set; }
+        private static readonly List<string> BlueColors = new()
+        {
+            "color_blue", "color_aquamarine", "color_dark_blue", "color_cyan", "color_light_cyan", "color_dark_cyan"
+        };
+        private Dictionary<int, int> GreyItems { get; set; }
+        private static readonly List<string> GreyColors = new()
+        {
+            "color_gray", "color_black", "color_poppyseed", "color_dark_gray"
+        };
+        private Dictionary<int, int> RedItems { get; set; }
+        private static readonly List<string> RedColors = new()
+        {
+            "color_red", "color_pink", "color_dark_pink", "color_salmon"
+        };
+        private Dictionary<int, int> YellowItems { get; set; }
+        private static readonly List<string> YellowColors = new()
+        {
+            "color_yellow", "color_gold", "color_sand", "color_dark_yellow"
+        };
+        private Dictionary<int, int> OrangeItems { get; set; }
+
+        private static readonly List<string> OrangeColors = new()
+        {
+            "color_orange", "color_dark_orange", "color_dark_brown", "color_brown", "color_copper"
+        };
+        public Dictionary<int, int> PurpleItems {get; set; }
+        private static readonly List<string> PurpleColors = new()
+        {
+            "color_purple", "color_dark_purple", "color_dark_pink", "color_pale_violet_red", "color_iridium"
+        };
+        public Dictionary<StardewItem, int> BerryItems {get; set; }
+        private static readonly string[] spring = new string[]{"spring"};
+        private static readonly string[] summer = new string[]{"summer"};
+        private static readonly string[] fall = new string[]{"fall"};
+
+        private class JunimoVendor
+        {
+            public PersistentStock JunimoPersistentStock {get; private set;}
+            public Dictionary<int, int> ColorItems {get; private set;}
+
+            public JunimoVendor(PersistentStock junimoPersistentStock, Dictionary<int, int> colorItems)
+            {
+                JunimoPersistentStock = junimoPersistentStock;
+                ColorItems = colorItems;
+            }
+        }
+
+        public JunimoShopGenerator(
+            ArchipelagoClient archipelago, ShopStockGenerator shopStockGenerator, StardewItemManager stardewItemManager)
+        {
+            _archipelago = archipelago;
+            _shopStockGenerator = shopStockGenerator;
+            _stardewItemManager = stardewItemManager;
+            RedPersistentStock = new PersistentStock();
```

Comment:
Actually, these stocks could be a local variable, then don't even need to exist at the class level, right?